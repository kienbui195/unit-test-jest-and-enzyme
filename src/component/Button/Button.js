import React from "react";

class Button extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            text: this.props.text
        }
        this.handleClick = this.handleClick.bind(this)
    }

    handleClick = () => {
        this.props.handleClick && this.props.handleClick()
    }

    render = () => {
        return (
            <button
                className={this.props.className || ''}
                onClick={this.handleClick}
                style={this.props.style || {}}>
                {this.state.text}
            </button>
        )
    }
}

export default Button;