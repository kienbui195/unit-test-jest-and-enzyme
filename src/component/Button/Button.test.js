import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';


//test with jest and react-test-renderer
describe('Button component', () => {
    it('renders without crashing', () => {
        renderer
            .create(<Button />)
    })

    it('renders with correct text  props', () => {
        const tree = renderer
            .create(<Button text='Click Me!' />)
            .toJSON();
        expect(tree).toMatchSnapshot()
    })

    it('renders with correct className prop', () => {
        const tree = renderer
            .create(<Button className='Hello' />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('calls handleClick prop when clicked', () => {
        const handleClick = jest.fn()
        const tree = renderer
            .create(<Button text='Click Me' className='Hello' handleClick={handleClick} />)
            .toJSON()
        tree.props.onClick()
        expect(handleClick).toHaveBeenCalledTimes(1)
    })
});
