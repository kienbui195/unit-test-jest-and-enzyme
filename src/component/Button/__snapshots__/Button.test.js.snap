// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
<button
  className="btn-primary"
  onClick={[Function]}
  style={Object {}}
>
  Click me
</button>
`;

exports[`Button component renders with correct className prop 1`] = `
<button
  className="Hello"
  onClick={[Function]}
  style={Object {}}
/>
`;

exports[`Button component renders with correct text  props 1`] = `
<button
  className=""
  onClick={[Function]}
  style={Object {}}
>
  Click Me!
</button>
`;
